### DISCUSSION TITLE
Issue with custom presets, some classes are lost when using empty presets array

### DISCUSSION THREAD

**Discussion by babakfp** [tailwindlabs/tailwindcss#12006] [Help] [answered]
Hi

Can someone check this out, please?

https://play.tailwindcss.com/BZ2Kfh4A5k?file=config

Useful information: #11428 

My TailwindCSS plugin: https://www.npmjs.com/package/tailwindcss-addons

The issue is when the plugin is used like this: https://babakfp.ir/docs/tailwindcss-addons#how-to-use-it, it results in an output like https://play.tailwindcss.com/BZ2Kfh4A5k?file=config, then the classes like `font-normal` are lost!

So, if the TailwindCSS config is something like this, then you are going to lose font-weight classes.

```js
module.exports = {
  presets: [
    {
      presets: [],
    },
  ],
}
```

Is this still something expected? Should I just remove `presets` when these configs all are set to `false`?

```js
presets: {
  flexGrid: false,
  fontWeightRegular: false,
  moreDefaultValues: false,
  screenSizeFix: false,
},
```

https://github.com/babakfp/tailwindcss-addons/blob/main/src/index.js

Thanks

**Comment by wongjn** [comment, marked as answer]
It is expected that when you have an empty `presets` array, you'd lose font-weight classes. [As per the documentation](https://tailwindcss.com/docs/presets#disabling-the-default-configuration):
> This will completely disable all of Tailwind’s defaults, so no colors, font families, font sizes, spacing values, etc. will be generated at all.
